src/coordinator/parser/promql/types.go

```diff
@@ -71,9 +71,16 @@ func NewOperator(opType promql.ItemType) (parser.Params, error) {
 
 // NewBinaryOperator creates a new binary operator based on the type
 func NewBinaryOperator(expr *promql.BinaryExpr, lhs, rhs parser.NodeID) (parser.Params, error) {
-	switch getOpType(expr.Op) {
+	matching := promMatchingToM3(expr.VectorMatching)
+
+	op := getOpType(expr.Op)
+	switch op {
 	case logical.AndType:
```

Comment:
how about 

case logical.AndType, logical.OrType, logical.UnlessType:

Reply:
Nice, didn't know we could do this